crates/schema-impl/src/key_extraction.rs

```diff
@@ -880,34 +882,34 @@ pub(crate) mod extract_impls {
         }
 
         // {a: "AA", b: "B"} and {a: "A", b: "AB"} are different keys!
-        #[test]
-        fn fields_are_correctly_separated() {
-            fn build_input_buf(str_1: &'static str, str_2: &'static str) -> Bytes {
-                // Prepare the key message
-                let mut key_msg = BytesMut::new();
-                prost::encoding::string::encode(1, &str_1.to_string(), &mut key_msg);
-                prost::encoding::string::encode(2, &str_2.to_string(), &mut key_msg);
-
-                // Prepare the root message (key is a nested message)
-                let mut out_msg = BytesMut::new();
-                encode_key(1, LengthDelimited, &mut out_msg);
-                encode_varint(key_msg.len() as u64, &mut out_msg);
-                out_msg.put(key_msg);
-
-                out_msg.freeze()
-            }
-
-            let root_key_field_number = 1;
-            let key_structure =
-                KeyStructure::Nested([(1, KeyStructure::Scalar), (2, KeyStructure::Scalar)].into());
-
-            let input_buf_a = build_input_buf("AA", "B");
-            let input_buf_b = build_input_buf("A", "AB");
-
-            assert_ne!(
-                root_extract(input_buf_a, root_key_field_number, &key_structure).unwrap(),
-                root_extract(input_buf_b, root_key_field_number, &key_structure).unwrap()
-            );
-        }
+        // #[test]
+        // fn fields_are_correctly_separated() {
+        //     fn build_input_buf(str_1: &'static str, str_2: &'static str) -> Bytes {
+        //         // Prepare the key message
+        //         let mut key_msg = BytesMut::new();
+        //         prost::encoding::string::encode(1, &str_1.to_string(), &mut key_msg);
+        //         prost::encoding::string::encode(2, &str_2.to_string(), &mut key_msg);
+        //
+        //         // Prepare the root message (key is a nested message)
+        //         let mut out_msg = BytesMut::new();
+        //         encode_key(1, LengthDelimited, &mut out_msg);
+        //         encode_varint(key_msg.len() as u64, &mut out_msg);
+        //         out_msg.put(key_msg);
+        //
+        //         out_msg.freeze()
+        //     }
+        //
+        //     let root_key_field_number = 1;
+        //     let key_structure =
+        //         KeyStructure::Nested([(1, KeyStructure::Scalar), (2, KeyStructure::Scalar)].into());
+        //
+        //     let input_buf_a = build_input_buf("AA", "B");
+        //     let input_buf_b = build_input_buf("A", "AB");
+        //
+        //     assert_ne!(
+        //         root_extract(input_buf_a, root_key_field_number, &key_structure).unwrap(),
+        //         root_extract(input_buf_b, root_key_field_number, &key_structure).unwrap()
+        //     );
+        // }
```

Comment:
I would be fine with removing the code that is commented now since we have the PR and the commit from which they can be reconstructed.